elements/ip/ipfilter.cc

```diff
@@ -1208,12 +1208,40 @@ IPFilter::Parser::parse_test(int pos, bool negated)
 }
 
 void
+merge_programs(Classification::Wordwise::Program &p1, Classification::Wordwise::Program &p2)
+{
+    int step_offset = p1.ninsn();
+    int jump_target = step_offset;
+    if (p1.ninsn() == 0) {
+        // p1 program is empty.
+        // If it has a valid target, we always jump to it. p2 is ignored.
+        // If target is invalid, proceed as normal. Resulting tree == p2.
+        int target = -p1.output_everything();
+
+        //check if target is valid
+        if (target != Classification::j_never + 1) {
```

Comment:
I know what `Classification::j_never` means and I know what `Classification::j_never + 1` means (it's referenced in elements/standard/classification.cc). But this is `(-Classification::j_never) + 1`, which is an integer overflow. I'm going to write this to what I think you meant but a test is really required.